packages/radar/src/hooks.ts

```diff
@@ -83,6 +85,11 @@ export const useRadar = <D extends Record<string, unknown>>({
         color: colorByKey[key],
     }))
 
+    const customLegends = legends.map(legend => ({
+        ...legend,
+        data: legend?.data?.map(d => ({ legendData, ...d })) || legendData,
```

Comment:
Why passing `legendData` (in `.map(d => ({ legendData, ...d }))`)? it seems a bit strange especially if passed for each datum.

Reply:
Misunderstanding caused a wrong legend data. 
Added unit test case.